proto/buf/alpha/registry/v1alpha1/schema.proto

```diff
@@ -0,0 +1,196 @@
+  //
+  // These files are a subset of the files in the "well-known types" module:
+  // https://buf.build/beta/wellknowntypes
+  //
+  // Specifically, the files that are omitted by default are the following:
+  //  * google/protobuf/any.proto
+  //  * google/protobuf/descriptor.proto
+  //  * google/protobuf/duration.proto
+  //  * google/protobuf/empty.proto
+  //  * google/protobuf/field_mask.proto
+  //  * google/protobuf/struct.proto
+  //  * google/protobuf/timestamp.proto
+  //  * google/protobuf/wrapper.proto
+  //
+  // These include all well-known types that have custom JSON formats and
+  // the types in "descriptor.proto", required for Protobuf reflection.
+  // Callers typically use compiled-in versions of these files.
+  bool include_well_known_imports = 6;
+
+  // If true, the returned schema will be expanded to include custom option
+  // definitions for options present on the requested schema elements.
+  bool include_custom_options = 7;
+
+  // If true, the returned schema will include all known extensions for any
+  // extendable messages among the requested schema elements.
+  bool include_extensions = 8;
+}
+
+message GetSchemaResponse {
+  // The resolved version of the schema. If the request included a reference to a commit,
+  // this value is the same as that commit. If the request included a reference to a tag,
+  // this is the commit version for that tag. If the request did not include any version,
+  // this is the latest version of the requested module.
+  //
+  // This version can be used in subsequent invocations in the only_if_not_version
+  // request field to poll for newer versions of the schema.
+  string version = 1;
+  // The schema, which is a set of file descriptors that include the requested elements
+  // and their dependencies.
+  //
+  // If the request included an only_if_not_version value, and the resolved version
+  // matched that version, this will be absent, indicating that the resolved version
+  // has not changed.
+  google.protobuf.FileDescriptorSet schema_files = 2;
+}
+
+message ReformatRequest {
+  // The module which contains the schema of interest. Required.
+  Module module = 1;
+  // The message type of the input data. Required.
+  string message_name = 2;
+
+  // The input data that is to be reformatted. Required.
+  bytes input_data = 3;
+
+  // The format of the input data. Required.
+  Format input_format = 4;
+  // If true, any unresolvable fields in the input are discarded. For
+  // formats other than FORMAT_BINARY, this means that the operation
+  // will fail if the input contains unrecognized field names. For
+  // FORMAT_BINARY, unrecognized fields can be retained and possibly
+  // included in the reformatted output (depending on the requested
+  // output format).
+  bool discard_unknown = 5;
+
+  oneof output_format {
+    BinaryOutputOptions output_binary = 6;
+    JSONOutputOptions output_json = 7;
+    TextOutputOptions output_text = 8;
+  }
```

Comment:
```
Failure: plugin go-api: plugin does not allow oneofs for request fields: OutputBinary
```
😭 🤮 

What can I do to work around this? This seems like a silly restriction. I know this is for our "de-structured" API clients... but I really don't like those anyway. Is there some way to just skip this file or service for that generator?

Not sure who to ask. Maybe @robbertvanginkel ?

Reply:
BSR-336 has a bit more details on this, I looked into supporting it once but its a disproportionate amount of work. Probably the quickest way to exempt it is in by https://github.com/bufbuild/buf/blob/e1cecca043cc170cd8e2c04bbcfb013f6ec54c78/make/buf/all.mk#L118 (although I think the cli only supports `--path` for includes?)

Reply:
Just promoted the link above to a draft PR: #1578.

But until all of that extra code gen and stuff is ripped out, I think the (admittedly kinda gross) workaround I pushed to this branch is okay. WDYT?